### PR TITLE
Hit or Miss game

### PR DESCRIPTION
**Pull request by joeraz**
Hello.  Hope I'm doing this right.

This will add another solitaire game, Hit or Miss ([https://en.wikipedia.org/wiki/Hit_or_Miss_(solitaire)](https://en.wikipedia.org/wiki/Hit_or_Miss_(solitaire))), to the PySolFC collection.  Although it's a rather simple, mechanical game, I personally always found it surprisingly appealing.

Note that I was able to run Travis successfully on this build, but I could not run Appveyor, likely due to the reported issue #195 .

### CODE REVIEW

**Comment by shlomif** [comment]
Hi Joe!

On Wed, 02 Dec 2020 19:41:13 -0800
"Joe R." <notifications@github.com> wrote:

> Hello.  Hope I&#39;m doing this right.
> 
> This will add another solitaire game, Hit or Miss
> ([https://en.wikipedia.org/wiki/Hit_or_Miss_(solitaire)](https://en.wikipedia.org/wiki/Hit_or_Miss_(solitaire))),
> to the PySolFC collection.  Although it&#39;s a rather simple, mechanical
> game, I personally always found it surprisingly appealing.
> 
> Note that I was able to run Travis successfully on this build, but I could
> not run Appveyor, likely due to the reported issue #195 . You can view,
> comment on, or merge this pull request online at:
> 
>   https://github.com/shlomif/PySolFC/pull/197
> 

Thanks for the PR! I'll try to take a look at it soonish.

> -- Commit Summary --
> 
>   * Added Hit or Miss game.
> 
> -- File Changes --
> 
>     A html-src/rules/hitormiss.html (15)
>     M pysollib/games/__init__.py (1)
>     A pysollib/games/hitormiss.py (162)
> 
> -- Patch Links --
> 
> https://github.com/shlomif/PySolFC/pull/197.patch
> https://github.com/shlomif/PySolFC/pull/197.diff
> 



-- 

Shlomi Fish       https://www.shlomifish.org/
https://youtu.be/GoEn1YfYTBM - Tiffany Alvord - “Fall Together”

When there is a PyPy booboo, then PyPy goes byebye!
    — https://www.shlomifish.org/humour.html#pypy_booboo

Please reply to list if it's a mailing list post - https://shlom.in/reply .

File: pysollib/games/hitormiss.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                self.max_rounds == VARIABLE_REDEALS):
+            return False
+        return True
+
+    def dealCards(self, sound=False):
+        game = self.game
+        old_state = game.enterState(game.S_FILL)
+        game.saveStateMove(2 | 16)  # for undo
+        if len(self.cards) > 0:
+            game.rank = game.rank + 1
```
Comment (shlomif) [inline pysollib/games/hitormiss.py:63]: @joeraz : please use `+= 1` here and elsewhere: https://careerkarma.com/blog/python-operator/

File: pysollib/games/hitormiss.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        layout.createText(stack, 'ne')
+        x, y = layout.XM+layout.XS, layout.YM+layout.YS
+        s.talon = HitOrMiss_Talon(x, y, self,
+                                  max_rounds=VARIABLE_REDEALS, num_deal=1)
+        layout.createText(s.talon, 'nw')
+
+        x += layout.XS
+        s.waste = HitOrMiss_Waste(x, y, self)
+        layout.createText(s.waste, 'ne')
+        x += layout.XS*1.1
```
Comment (shlomif) [inline pysollib/games/hitormiss.py:115]: delimit the asterisk with spaces `x += layout.XS * 1.1`. also applies to other binary infix operators

**Review by shlomif — COMMENTED** [review]
@jo

**Comment by shlomif** [comment]
@joeraz : please revise the code per my review comments and squash the branch.

File: pysollib/games/hitormiss.py (hunk 1 above)
Comment (joeraz) [reply pysollib/games/hitormiss.py]: Done.

File: pysollib/games/hitormiss.py (hunk 2 above)
Comment (joeraz) [reply pysollib/games/hitormiss.py]: Done.

**Comment by joeraz** [comment]
@shlomif Updated and squashed.  Thanks.

**Comment by shlomif** [comment]
On Thu, 03 Dec 2020 14:46:24 -0800
"Joe R." <notifications@github.com> wrote:

> @shlomif Updated and squashed.  Thanks.
> 

Thanks Joe! I'll try to review the PR (and test it) soon, but note that I have
other priorities. Other admins with push permissions are free to merge it as
well.



-- 

Shlomi Fish       https://www.shlomifish.org/
What Makes Software Apps High Quality -  https://shlom.in/sw-quality

Some people grow older and wiser. Not I. I grow older and more foolish.

Please reply to list if it's a mailing list post - https://shlom.in/reply .

**Comment by shlomif** [comment]
Merged now, thanks! Sorry it took me so long.

**Pull request merged**